examples/doublemach-mpi.py

```diff
@@ -0,0 +1,259 @@
+
+    Input required:
+        doubleMach.msh (read existing mesh)
+
+    This routine will generate a new grid if it does
+    not find the grid file (doubleMach.msh).
+    """
+    from meshmode.mesh.io import (
+        read_gmsh,
+        generate_gmsh,
+        ScriptSource,
+    )
+    import os
+    meshfile = "doubleMach.msh"
+    if os.path.exists(meshfile) is False:
+        mesh = generate_gmsh(
+            ScriptSource("""
+                x0=1.0/6.0;
+                setsize=0.025;
+                Point(1) = {0, 0, 0, setsize};
+                Point(2) = {x0,0, 0, setsize};
+                Point(3) = {4, 0, 0, setsize};
+                Point(4) = {4, 1, 0, setsize};
+                Point(5) = {0, 1, 0, setsize};
+                Line(1) = {1, 2};
+                Line(2) = {2, 3};
+                Line(5) = {3, 4};
+                Line(6) = {4, 5};
+                Line(7) = {5, 1};
+                Line Loop(8) = {-5, -6, -7, -1, -2};
+                Plane Surface(8) = {8};
+                Physical Surface('domain') = {8};
+                Physical Curve('ic1') = {6};
+                Physical Curve('ic2') = {7};
+                Physical Curve('ic3') = {1};
+                Physical Curve('wall') = {2};
+                Physical Curve('out') = {5};
+        """, "geo"), force_ambient_dim=2, dimensions=2, target_unit="M")
+    else:
+        mesh = read_gmsh(meshfile, force_ambient_dim=2)
+
+    return mesh
+
+
+@mpi_entry_point
+def main(ctx_factory=cl.create_some_context):
+    """Drive the example."""
+    cl_ctx = ctx_factory()
+    queue = cl.CommandQueue(cl_ctx)
+    actx = PyOpenCLArrayContext(
+        queue, allocator=cl_tools.MemoryPool(cl_tools.ImmediateAllocator(queue))
+    )
+
+    dim = 2
+    order = 3
+    # Too many steps for CI
+    # t_final = 1.0e-2
+    t_final = 1.0e-3
+    current_cfl = 0.1
+    current_dt = 1.0e-4
+    current_t = 0
+    eos = IdealSingleGas()
+    initializer = DoubleMachReflection(dim)
+    casename = "doubleMach"
+    from grudge import sym
+
+    boundaries = {
+        sym.DTAG_BOUNDARY("ic1"): PrescribedBoundary(initializer),
+        sym.DTAG_BOUNDARY("ic2"): PrescribedBoundary(initializer),
+        sym.DTAG_BOUNDARY("ic3"): PrescribedBoundary(initializer),
+        sym.DTAG_BOUNDARY("wall"): AdiabaticSlipBoundary(),
+        sym.DTAG_BOUNDARY("out"): AdiabaticSlipBoundary(),
+    }
+    constant_cfl = False
+    nstatus = 10
+    nviz = 100
+    rank = 0
+    checkpoint_t = current_t
+    current_step = 0
+    timestepper = rk4_step
+
+    s0 = -6.0
+    kappa = 1.0
+    alpha = 2.0e-2
+    from mpi4py import MPI
+
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank
+
+    gen_grid = partial(get_doublemach_mesh)
+
+    local_mesh, global_nelements = create_parallel_grid(comm, gen_grid)
+
+    local_nelements = local_mesh.nelements
+
+    discr = EagerDGDiscretization(actx, local_mesh, order=order,
+                                  mpi_communicator=comm)
```

Comment:
Hmm, this looks like it would require more changes. In particular it looks like `DiscretizationCollection` does not have nodes associated with it. I think all the examples would need similar modifications?

Reply:
I think this is a drop-in replacement. It should *just work*.

Reply:
I could be donig something wrong (or my branch is quite working), but I get that nodes doesn't exist
```Traceback (most recent call last):
  File "/usr/workspace/hagen7/mirgecom/quartz/mirgetesting/emirge/miniforge3/envs/ceesd/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/workspace/hagen7/mirgecom/quartz/mirgetesting/emirge/miniforge3/envs/ceesd/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/workspace/hagen7/mirgecom/quartz/mirgetesting/emirge/miniforge3/envs/ceesd/lib/python3.8/site-packages/mpi4py/__main__.py", line 7, in <module>
    main()
  File "/usr/workspace/hagen7/mirgecom/quartz/mirgetesting/emirge/miniforge3/envs/ceesd/lib/python3.8/site-packages/mpi4py/run.py", line 196, in main
    run_command_line(args)
  File "/usr/workspace/hagen7/mirgecom/quartz/mirgetesting/emirge/miniforge3/envs/ceesd/lib/python3.8/site-packages/mpi4py/run.py", line 47, in run_command_line
    run_path(sys.argv[0], run_name='__main__')
  File "/usr/workspace/hagen7/mirgecom/quartz/mirgetesting/emirge/miniforge3/envs/ceesd/lib/python3.8/runpy.py", line 265, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/usr/workspace/hagen7/mirgecom/quartz/mirgetesting/emirge/miniforge3/envs/ceesd/lib/python3.8/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/workspace/hagen7/mirgecom/quartz/mirgetesting/emirge/miniforge3/envs/ceesd/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "./doublemach-mpi.py", line 257, in <module>
    main()
  File "/usr/WS2/hagen7/mirgecom/quartz/mirgetesting/emirge/mirgecom/mirgecom/mpi.py", line 154, in wrapped_func
    func(*args, **kwargs)
  File "./doublemach-mpi.py", line 161, in main
    nodes = thaw(actx, discr.nodes())
AttributeError: 'DiscretizationCollection' object has no attribute 'nodes'
```

Reply:
I stand corrected.  I have no idea.  I guess we need to pull the volume discr out somehow.

Reply:
Yeah, I vote we hold off on this.  It looks like this might entail sweeping changes to all of *MIRGE-Com*.

Reply:
OK I see what happened. For now, don't apply this change; keep `EagerDGDiscretization` as is. This will need some plumbing